IQSS/dataverse-frontend
# Feat/484 Collection page results of all types

## What this PR does / why we need it:
- Integrates the new `getCollectionItems` use case to display collections, datasets and files in the collection page.
- Creates the search box UI and functionality fully integrated with the use case.
- Creates the checkboxes UI to filter by collection, datasets and files fully integrated with the use case.

## Which issue(s) this PR closes:

- Closes #484 

## Special notes for your reviewer:

## Suggestions on how to test this:
**Step 1: Run the Development Environment**
1. Execute `npm i`.
2. Navigate with `cd packages/design-system && npm run build`.
3. Return with `cd ../../`.
4. Ensure you have a `.env` file similar to `.env.example`, with the variable `VITE_DATAVERSE_BACKEND_URL=http://localhost:8000`.
5. Navigate with `cd dev-env`.
6. Start the environment using `./run-env.sh unstable`.
7. To verify the environment, visit <http://localhost:8000> and check your local Dataverse installation.

**Step 2: Test the feature**
First of all create collections, datasets and files within some of those datasets.

- Filter results by type with the checkboxes, validate that only those type of items are being displayed and that the URL is synced with the types.
- Make a search, validate that the results are shown and the URL is synced with all the types and the query param. (*Note: to be sure that an api endpoint call is made with the search value, open the network tab and filter by 'search?', check that the payload has te correct search value)
- Navigate to a collection and see that the results are according to that collection.
- If there are more items than the ones displayed, scroll to the bottom to load more items.
- Go back and forward with the browser navigation buttons and validate the results are show correctly.
- Go to the home page and search a value from there and validate that the results according to it are shown.

## Does this PR introduce a user interface change? If mockups are available, please link/include them here: Yes.

https://github.com/user-attachments/assets/40c0a9c1-a405-4601-b32e-43622f5e6ce0

https://github.com/user-attachments/assets/4d79948f-6242-499e-bc71-740fa67a6a06

## Is there a release notes update needed for this change?:
No

## Additional documentation:
No
